include/mlir/IR/AffineMap.h

```diff
@@ -47,9 +47,10 @@ class AffineMap {
 public:
   using ImplType = detail::AffineMapStorage;
 
-  AffineMap() : map(nullptr) {}
-  explicit AffineMap(ImplType *map) : map(map) {}
-  AffineMap(const AffineMap &other) : map(other.map) {}
+  AffineMap() : map(nullptr), context(nullptr) {}
+  explicit AffineMap(ImplType *map, MLIRContext *context)
+      : map(map), context(context) {}
```

Comment:
Why not store the context in the AffineMapStorage? Otherwise you are duplicating per instance, and AffineMap is now two words instead of one.

Reply:
Good point! Yeah, it makes more sense. Done. Thanks!